lib/dockerregistry/checks.go

```diff
@@ -0,0 +1,133 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inventory
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	gogit "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+// ImageRemovalCheck implements the PreCheck interface and checks against
+// pull requests that attempt to remove any images from the promoter manifests.
+type ImageRemovalCheck struct {
+	GitRepoURL     string
+	MasterSHA      plumbing.Hash
+	PullRequestSHA plumbing.Hash
+}
+
+// MKRealImageRemovalCheck returns an instance of ImageRemovalCheck.
+func MKRealImageRemovalCheck(gitRepoURL string) *ImageRemovalCheck {
+	masterSHAStr := os.Getenv("PULL_BASE_SHA")
+	pullRequestSHAStr := os.Getenv("PULL_PULL_SHA")
+	masterSHA := plumbing.NewHash(masterSHAStr)
+	pullRequestSHA := plumbing.NewHash(pullRequestSHAStr)
+	return &ImageRemovalCheck{
+		gitRepoURL,
+		masterSHA,
+		pullRequestSHA,
+	}
+}
+
+// Run executes ImageRemovalCheck on a set of promotion edges.
+// Returns an error if the pull request removes images from the
+// promoter manifests.
+func (check *ImageRemovalCheck) Run(
+	edges map[PromotionEdge]interface{},
+) error {
+	var err error
+	// Get the Git repo the current working directory
+	r, err := gogit.PlainOpen(check.GitRepoURL)
+	if err != nil {
+		return fmt.Errorf("Could not open the Git repo: %v", err)
+	}
+
+	// Check out the master branch of the current Git repo to get the
+	// master branch's state of the promoter manifests
+	w, err := r.Worktree()
+	if err != nil {
+		return fmt.Errorf("Could not create Git worktree: %v", err)
+	}
+	err = w.Checkout(&gogit.CheckoutOptions{
+		Hash:  check.MasterSHA,
+		Force: true,
+	})
+	if err != nil {
+		return fmt.Errorf("Could not checkout the master branch of the Git"+
+			" repo: %v", err)
+	}
+
+	// Parse the manifests and generate the promotion edges
+	mfests, err := ParseThinManifestsFromDir(check.GitRepoURL)
+	if err != nil {
+		return fmt.Errorf("Could not parse manifests from the directory: %v",
+			err)
+	}
+	masterEdges, err := ToPromotionEdges(mfests)
+	if err != nil {
+		return fmt.Errorf("Could not generate promotion edges from promoter"+
+			" manifests: %v", err)
+	}
+
+	// Reset the working Git repo by checking out the pull request
+	err = w.Checkout(&gogit.CheckoutOptions{
+		Hash:  check.PullRequestSHA,
+		Force: true,
+	})
+	if err != nil {
+		return fmt.Errorf("Could not checkout the pull request branch of the"+
+			" Git repo %v: %v",
+			check.GitRepoURL, err)
+	}
+
+	return check.CompareEdges(edges, masterEdges)
```

Comment:
I would prefer to flip the order of arguments, and also drop `Edges` from the name. So the call should look like `check.Compare(masterEdges, edges)`.